Blurhash.ImageSharp/Encoder.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using System.Net.Mime;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Blurhash.Core;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+
+namespace Blurhash.ImageSharp
+{
+    public class Encoder : CoreEncoder 
+    {
+        /// <summary>
+        /// Encodes a picture into a Blurhash string
+        /// </summary>
+        /// <param name="image">The picture to encode</param>
+        /// <param name="componentsX">The number of components used on the X-Axis for the DCT</param>
+        /// <param name="componentsY">The number of components used on the Y-Axis for the DCT</param>
+        /// <returns>The resulting Blurhash string</returns>
+        public string Encode(Image<Rgb24> image, int componentsX, int componentsY)
+        {
+            return CoreEncode(ConvertBitmap(image), componentsX, componentsY);
+        }
+        
+        /// <summary>
+        /// Encodes a picture into a Blurhash string
+        /// </summary>
+        /// <param name="image">The picture to encode</param>
+        /// <param name="componentsX">The number of components used on the X-Axis for the DCT</param>
+        /// <param name="componentsY">The number of components used on the Y-Axis for the DCT</param>
+        /// <returns>The resulting Blurhash string</returns>
+        public string Encode(Image<Rgba32> image, int componentsX, int componentsY)
+        {
+            return CoreEncode(ConvertBitmap(image), componentsX, componentsY);
+        }
+
+        /// <summary>
+        /// Converts the given bitmap to the library-independent representation used within the Blurhash-core
+        /// </summary>
+        /// <param name="sourceBitmap">The bitmap to encode</param>
+        internal static Pixel[,] ConvertBitmap<T>(Image<T> sourceBitmap) where T : struct, IPixel<T>
+        {
+            var width = sourceBitmap.Width;
+            var height = sourceBitmap.Height;
+            var bytesPerPixel = sourceBitmap.PixelType.BitsPerPixel / 8;
+            var stride = width * 3;
+
+            var result = new Pixel[width, height];
+            
+            for (int y = 0; y < height; y++)
+            {
+                var rgbValues = MemoryMarshal.AsBytes(sourceBitmap.GetPixelRowSpan(y));
+
+                var index = stride;
+
+                for (var x = 0; x < width; x++)
+                {
+                    result[x, y].Red = MathUtils.SRgbToLinear(rgbValues[index]);
```

Comment:
You will get incorrect results here with `Image<Rgba32>` or any other pixel type.

Reply:
I think I'll get the correct behaviour for `Rgba32` & `Rgb24`, basically and pixel where the first 24bits are RGB. The public methods guard for the supported types.
Have I understood your concern? Or have is there something that wont work?

Reply:
> I think I'll get the correct behaviour for Rgba32 & Rgb24

Yeah that's actually true, I missed `bytesPerPixel`. However, you'll get the colors wrong with `Bgr24` or `Argb32` or anything else.

Reply:
I've added a guard now to constrain the pixel type

Reply:
Missed that as well ^^

Reply:
rgbValues[index] throws outofbounds error

Reply:
@slang25 Do you reckon it would be possible to give me write access to your fork? I have ideas to polish this.

Reply:
Done 😃